flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/CodeGenUtils.scala

```diff
@@ -477,40 +477,26 @@ object CodeGenUtils {
         case Some(writer) =>
           // use writer to set field
           val writeField = binaryWriterWriteField(ctx, indexTerm, fieldTerm, writer, fieldType)
-          if (ctx.nullCheck) {
-            s"""
-               |${fieldExpr.code}
-               |if (${fieldExpr.nullTerm}) {
-               |  ${binaryWriterWriteNull(indexTerm, writer, fieldType)};
-               |} else {
-               |  $writeField;
-               |}
-             """.stripMargin
-          } else {
-            s"""
-               |${fieldExpr.code}
-               |$writeField;
-             """.stripMargin
-          }
+          s"""
```

Comment:
Now in theory, at every location where you're removing the `ctx.nullCheck`, you should check if the type of the `fieldExpr` is nullable or not.

Problem is that a lot of code doesn't care about nullability... Maybe do you wanna give it a try and see if it works?

Reply:
You mean, to have the similar code but instead of checking the `ctx.nullCheck`, check the nullability of the field and if it's not null then avoid the extra code?

Reply:
yep that's what i mean

Reply:
Please check again, I have squashed my changes with the second commit, apologies for that